src/Components/Scheduling/SchedulingColumn.tsx

```diff
@@ -229,11 +229,11 @@ const placeStaggeredBlocks = (blocks: APICourseBlock[], filter: any) => {
           { preparedBlock.collisions_left.map((collision: number, idx: number) => {
             collisionReferences.get(bidx).push(collision);
             if (blocks[bidx].start_time.getTime() === blocks[collision].start_time.getTime() && blocks[bidx].end_time.getTime() === blocks[collision].end_time.getTime()) {
-              return < SchedulingBlock spacer={true} size={'auto'} visible={filter[blocks[collision].course_number]} linkIDs={[]} key={`left-spacer-${idx}`}/>
+              return < SchedulingBlock spacer={true} size={'auto'} visible={filter.get(blocks[collision].course_number)!} linkIDs={[]} key={`left-spacer-${idx}`}/>
             }
-            return < SchedulingBlock spacer={true} size={(ratios.has(collision))? `calc(${calcRatio(collision)}% - 4px)` : undefined} visible={filter[blocks[collision].course_number]} linkIDs={[]} key={`left-spacer-${idx}`}/>
+            return < SchedulingBlock spacer={true} size={(ratios.has(collision))? `calc(${calcRatio(collision)}% - 4px)` : undefined} visible={filter.get(blocks[collision].course_number)!} linkIDs={[]} key={`left-spacer-${idx}`}/>
           })}
-          < SchedulingBlock size={calcSelf()} course_instance={block} visible={filter[block.course_number]} linkIDs={block.scheduled} key={`block-${JSON.stringify(block)}`}/>
+          < SchedulingBlock size={calcSelf()} course_instance={block} visible={filter.get(block.course_number)!} linkIDs={block.scheduled} key={`block-${JSON.stringify(block)}`}/>
```

Comment:
This is the same suffix `!`. I still am not sure what this syntax means. If you resolve my other comment, resolve this one too.

Reply:
I cant find any information on it, maybe it is logical not but on the other side cause javascript

Reply:
Force unwrap :D